src/meta/src/hummock/manager/compaction/compaction_group_schedule.rs

```
@@ -711,3 +546,340 @@ impl HummockManager {
+                    let ret = self
+                        .move_state_tables_to_dedicated_compaction_group(
+                            group.group_id,
+                            &table_ids,
+                            0,
+                        )
+                        .await;
+                    match ret {
+                        Ok(split_result) => {
+                            tracing::info!(
+                                "split_huge_compaction_group success {:?}",
+                                split_result
+                            );
+                            self.metrics
+                                .split_compaction_group_count
+                                .with_label_values(&[&group.group_id.to_string()])
+                                .inc();
+                            return;
+                        }
+                        Err(e) => {
+                            tracing::error!(
+                                error = %e.as_report(),
+                                "failed to split_huge_compaction_group table {:?} from group-{}",
+                                table_ids,
+                                group.group_id
+                            );
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    pub async fn try_merge_compaction_group(
+        &self,
+        table_write_throughput: &HashMap<u32, VecDeque<u64>>,
+        group: &TableGroupInfo,
+        group_config: CompactionGroup,
+        next_group: &TableGroupInfo,
+        _next_group_config: CompactionGroup,
+        checkpoint_secs: u64,
+        created_tables: &HashSet<u32>,
+    ) -> Result<()> {
+        // TODO: remove this check after refactor group id
+        if group.group_id == StaticCompactionGroupId::StateDefault as u64
+            && next_group.group_id == StaticCompactionGroupId::MaterializedView as u64
+        {
+            return Ok(());
```

Should we return an error instead of ok here? With logics in timer_task.rs L621-630, I think returning error is more suitable because it will advance the left group as well.